src/commands/vault/add/processor.ts

```diff
@@ -0,0 +1,113 @@
+import config from "adapters/config"
+import { CommandInteraction } from "discord.js"
+import { GuildIdNotFoundError } from "errors"
+import { MessageEmbed } from "discord.js"
+import { APIError } from "errors"
+import { getEmoji, msgColors } from "utils/common"
+import { getErrorEmbed } from "ui/discord/embed"
+
+export async function runAddTreasurer({
+  i,
+  guildId,
+}: {
+  i: CommandInteraction
+  guildId?: string | null
+}) {
+  if (!guildId) {
+    throw new GuildIdNotFoundError({ message: i })
+  }
+
+  const user = i.options.getUser("user")
+  if (!user) {
+    return {
+      messageOptions: {
+        embeds: [
+          getErrorEmbed({
+            description: "Invalid user. Please choose another one!",
+          }),
+        ],
+      },
+    }
+  }
+
+  const vaultName = i.options.getString("name") ?? ""
+  const res = await config.createAddTreasureRequest({
```

Comment:
Xài fetchJson lấy ra status

```suggestion
  const { ok, data, status } = await config.createAddTreasureRequest({...})
```